web/src/main/java/no/nav/modiapersonoversikt/rest/persondata/PersondataService.kt

```diff
@@ -151,14 +165,18 @@ class PersondataServiceImpl(
                 break
             }
         }
-        return PersondataResult.runCatching(InformasjonElement.NORG_NAVKONTOR) {
-            norgApi
-                .finnNavKontor(gt, diskresjonskode)
-                ?.enhetId
-        }
-            .map(InformasjonElement.NORG_KONTAKTINFORMASJON) {
-                it?.let { enhetId -> norgApi.hentKontaktinfo(EnhetId(enhetId)) }
+        return if (erGyldigGT(gt)) {
```

Comment:
For å slippe å iterere over `adressebeskyttelse` kan sjekken flyttes høyere opp.

F.eks kan man utvide `PersondataResult` med en funksjon alà `fun <S> flatMap(block: (t: PersondataResult<T>) -> PersondataResult<S>): PersondataResult<S>` og deretter bruke den.

```kotlin
        val gt: String = geografiskeTilknytning
            .flatMap {
                if (erGyldigGT(it.getOrNull())) {
                    it
                } else {
                    PersondataResult.NotRelevant()
                }
            }
            .fold(
                onSuccess = { it ?: "" },
                onNotRelevant = { null },
                onFailure = { _, _ -> "" }
            ) ?: return PersondataResult.NotRelevant()
```